lib/python/Pyxsim/__init__.py

```diff
@@ -71,6 +72,44 @@ def _build(xe_path, build_config=None, env={}, do_clean=False, clean_only= False
     return (success, output)
 
 
+def run_on_simulator_(xe, tester=None, simthreads=[], **kwargs):
+    
+    do_xe_prebuild = kwargs.get('do_xe_prebuild', False)
+    capfd = kwargs.get('capfd', None)
+    kwargs.pop("capfd")
```

Comment:
pop will raise an exception if "capfd" is not a dictionary key, so you could call pop with a default. Or since pop returns the value anyway, you could just do pop instead of get and then pop.